### PR TITLE
Adds matrix and path editing to the variable viewer

### PR DESCRIPTION
**Pull request by Crazylemon64**
I wasn't able to find a method of simply VIEWING a matrix, however - info would be welcome!

- [ ] Add a means of actually viewing a matrix instead of just overwriting it
- [ ] Refactor variable editing code to separate out the special-case edits (`stat` and `luminosity`) into a different proc, and push special variable behavior there

:cl:Crazylemon
rscadd: People with VAREDIT can now write matrix variables
rscadd: People with VAREDIT can now modify path variables
tweak: Refactors the variable editor code somewhat.
/:cl:

### CODE REVIEW

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -424,4 +424,23 @@ proc/tg_text2list(text, glue=",", assocglue=";")
 		if(temp <= 16)
 			. = 0
 		else
-			. = max(0, min(255, 138.5177312231 * log(temp - 10) - 305.0447927307))
\ No newline at end of file
+			. = max(0, min(255, 138.5177312231 * log(temp - 10) - 305.0447927307))
+
+//Argument: Give this a space-separated string consisting of 6 numbers. Returns null if you don't
+/proc/text2matrix(var/matrixtext)
```
Comment (tigercat2000) [inline code/__HELPERS/type2type.dm]: why only 6 numbers? aren't matrixes normally using 9/16/25 layouts?

Comment (Crazylemon64) [inline code/__HELPERS/type2type.dm]: The BYOND `matrix` proc accepts only 6 arguments - unless there's undocumented behavior I'm unaware of. This results in a matrix shaped as follows:

a d 0
b e 0
c f 1

Comment (tigercat2000) [inline code/__HELPERS/type2type.dm]: Further investigation reveals that BYOND only uses the `matrix()` proc for transform matrices, and uses `list()`s for color matricies. So this is correct for it's purpose.

**Comment by Crazylemon64** [comment]
Matrix editing still could use some work, but this is ready for merging as-is.

**Pull request merged**